follow.go

```diff
@@ -121,3 +122,84 @@ func following(ctx context.Context, c *client, userID string, opt ...*FollowOpti
 
 	return &f, nil
 }
+
+func postFollowing(ctx context.Context, c *client, userID string, targetUserID string) (*PostFollowingResponse, error) {
+	if userID == "" {
+		return nil, errors.New("post following by userID: userID parameter is required")
+	}
+	ep := fmt.Sprintf(postFollowingURL, userID)
+
+	if targetUserID == "" {
+		return nil, errors.New("post following by targetUserID: targetUserID parameter is required")
+	}
+	body := &FollowingBody{
+		TargetUserID: targetUserID,
+	}
+	jsonStr, err := json.Marshal(body)
+	if err != nil {
+		return nil, errors.New("post following: can not marshal")
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, ep, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		return nil, fmt.Errorf("post following new request with ctx: %w", err)
+	}
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.bearerToken))
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("post following response: %w", err)
+	}
+	defer resp.Body.Close()
+
+	var postFollowing PostFollowingResponse
+	if err := json.NewDecoder(resp.Body).Decode(&postFollowing); err != nil {
+		return nil, fmt.Errorf("post following decode: %w", err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		return &postFollowing, &HTTPError{
+			APIName: "post following",
+			Status:  resp.Status,
+			URL:     req.URL.String(),
+		}
+	}
+
+	return &postFollowing, nil
+}
+
+func undoFollowing(ctx context.Context, c *client, sourceUserID string, targetUserID string) (*UndoFollowingResponse, error) {
+	if sourceUserID == "" {
+		return nil, errors.New("undo following by sourceUserID: sourceUserID parameter is required")
```

Comment:
I think this error message isn't good.
This is because some of the error messages are duplicated.